modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/DataStreamOptionsIT.java

```diff
@@ -137,11 +148,17 @@ public void testEnableDisableFailureStore() throws IOException {
                   }
                 }""");
             assertAcknowledged(client().performRequest(enableRequest));
-            assertFailureStore(true, 1);
+            setDataStreamFailureStoreClusterSetting(DATA_STREAM_NAME);
+            assertDataStreamOptions(true);
+            assertFailureStoreValuesInGetDataStreamResponse(true, 1);
+            assertRedirectsDocWithBadMappingToFailureStore();
+            setDataStreamFailureStoreClusterSetting("does-not-match-failure-data-stream"); // should have no effect as enabled in options
             assertDataStreamOptions(true);
+            assertFailureStoreValuesInGetDataStreamResponse(true, 1);
+            assertRedirectsDocWithBadMappingToFailureStore();
```

Comment:
> Nit: should we set setDataStreamFailureStoreClusterSetting("does-not-match-failure-data-stream"); a few lines earlier and perform the checks only once?

Right now, we're testing all combinations of [enabled in options, disabled in options, not specified in options] ⊗ [enabled by settings, not enabled by settings]. I guess you're saying that we could reduce that by not bothering to check enabled in options and enabled by settings when we expect the same outcome with enabled in settings and not enabled by settings and the latter is the 'stronger' test? I see what you mean — but, on the other hand, why _not_ test all the combinations? Surely a fraction of a second on the test execution time isn't a problem?

> The difference between having the cluster setting set to a non-matching value or null shouldn't influence this test, so we demonstrate that by this randomisation.

I'd actually rather test both a non-matching value and a null value, rather than randomly testing one of the two. As discussed in a Slack thread a while ago, I think we over-used randomization in tests a bit, and if we care about two behaviours then we should just always test both of them. According to my highly unscientific benchmarking, adding tests for the null setting with all three options values costs us about 75ms, which I think is okay. I'll push a commit with that now.

Reply:
Ok, no strong feelings about this.